code/game/gamemodes/devil/devilinfo.dm

```diff
@@ -435,5 +435,6 @@ var/global/list/lawlorify = list (
 	soulCounter = SOULVALUE
 	var/obj/screen/devil/soul_counter/K = new()
 	K.update()
+	var/mob/living/carbon/D = new()
```

Comment:
Aright, I'll help you out some.
Remove everything in this proc, replace it with
owner.current.handle_devil(SOULVALUE)